src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs

```diff
@@ -148,6 +148,9 @@ private async Task<(Uri url, CancellationToken hostExitToken)> StartIISExpressAs
                         {
                             Logger.LogTrace($"Config File Content:{Environment.NewLine}===START CONFIG==={Environment.NewLine}{{configContent}}{Environment.NewLine}===END CONFIG===", serverConfig);
                         }
+                        var hostingModel = DeploymentParameters.HostingModel.ToString();
+                        serverConfig.Replace("[HostingModel]", hostingModel);
```

Comment:
Why not just chain it here: https://github.com/aspnet/Hosting/blob/05fd382b93cb51668d1d8858e79292601ca71b18/src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs#L138-L141

Reply:
The contentRoot and port are inputs to StartIISExpressAsync. HostingModel isn't guaranteed to be set. I will add a check to see if HostingModel exists before setting it though.